FWCore/TestProcessor/src/TestProcessor.cc

```diff
@@ -563,26 +559,22 @@ namespace edm {
 
       LumiTransitionInfo transitionInfo(*lumiPrincipal_, es, nullptr);
 
-      std::vector<edm::SubProcess> emptyList;
       {
         using Traits = OccurrenceTraits<LuminosityBlockPrincipal, BranchActionGlobalBegin>;
         FinalWaitingTask globalWaitTask{taskGroup_};
-        beginGlobalTransitionAsync<Traits>(
-            WaitingTaskHolder(taskGroup_, &globalWaitTask), *schedule_, transitionInfo, serviceToken_, emptyList);
+        schedule_->processOneGlobalAsync<Traits>(
+            WaitingTaskHolder(taskGroup_, &globalWaitTask), transitionInfo, serviceToken_);
         globalWaitTask.wait();
       }
       {
         //To wait, the ref count has to be 1+#streams
         FinalWaitingTask streamLoopWaitTask{taskGroup_};
 
         using Traits = OccurrenceTraits<LuminosityBlockPrincipal, BranchActionStreamBegin>;
-
-        beginStreamsTransitionAsync<Traits>(WaitingTaskHolder(taskGroup_, &streamLoopWaitTask),
-                                            *schedule_,
-                                            preallocations_.numberOfStreams(),
-                                            transitionInfo,
-                                            serviceToken_,
-                                            emptyList);
+        for (unsigned int i = 0; i < preallocations_.numberOfStreams(); ++i) {
+          schedule_->processOneStreamAsync<Traits>(
+              WaitingTaskHolder(taskGroup_, &streamLoopWaitTask), i, transitionInfo, serviceToken_);
+        }
```

Comment:
I find the replacement of `beginGlobalTransitionAsync()` with `schedule_->processOneGlobalAsync()` fine as such, although one could make an argument against along "if we ever want to have a customization point there, we need to add the `beginGlobalTransitionAsync()` back".

I'm less certain about the replacement of `beginStreamsTransitionAsync()` with the loop over streams calling `schedule_->processOneStreamAsync()`. Or in other words, I'd find some value in keeping the loop over streams abstracted in `beginStreamsTransitionAsync()`.

Further thoughts? @wddgit @Dr15Jones

Reply:
What ever we do we should keep `EventProcessor` and `TestProcessor` using the same code, preferable by sharing the same routines.